### PR TITLE
Make cluster_backup_image config optional

### PR DESCRIPTION
**Pull request by infoverload**
## Summary of the changes / Why this is an improvement

https://github.com/crate/crate-operator/issues/109

## Checklist

 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed

### CODE REVIEW

**Comment by lukasbals** [comment]
I think the documentation (https://github.com/crate/crate-operator/blob/master/docs/source/configuration.rst) also needs to be updated. 🤔

File: crate/operator/config.py [hunk 1]
```diff
@@ -45,7 +45,7 @@ class Config:
     CLOUD_PROVIDER: Optional[CloudProvider] = None
 
     #: The Docker image that contians scripts to run cluster backups.
```
Comment (chaudum) [inline crate/operator/config.py:47]:
```suggestion
    #: The Docker image that contains scripts to run cluster backups.
```

File: crate/operator/main.py [hunk 2]
```diff
@@ -360,25 +360,30 @@ async def cluster_create(
...
-                prometheus_port,
-                spec["backups"],
-                image_pull_secrets,
-                "ssl" in spec["cluster"],
-                logger,
-            ),
-            id="backup",
-        )
+        if config.CLUSTER_BACKUP_IMAGE is None:
+            logger.warning(
```
Comment (chaudum) [inline crate/operator/main.py:364]: I think the log level `WARN` is too high for this information, I'd rather use `INFO`.

**Review by chaudum — COMMENTED** [review]
Please also add a changelog entry and update the documentation :)

File: CHANGES.rst [hunk 3]
```diff
@@ -42,6 +42,8 @@ Unreleased
 * Fixed a bug that would prevent the version of the Docker image of the
   ``mkdir-heapdump`` init container to be updated when a cluster is upgraded.
 
+* Made CLUSTER_BACKUP_IMAGE config optional.
```
Comment (lukasbals) [inline CHANGES.rst:45]: Usually, we add the new changelog entries at the top of the unreleased changes 🤔  Not sure how strict this is ...

**Review by lukasbals — APPROVED** [review]
Otherwise LGTM. Good work 🚀

File: CHANGES.rst (hunk 3 above)
Comment (infoverload) [reply CHANGES.rst]: I made the change.

File: CHANGES.rst [hunk 4]
```diff
@@ -5,16 +5,18 @@ Changelog
 Unreleased
 ----------
 
+* Made CLUSTER_BACKUP_IMAGE config optional.
```
Comment (chaudum) [inline CHANGES.rst:8]:
```suggestion
* Made ``CLUSTER_BACKUP_IMAGE`` configuration parameter optional to remove dependency on external Docker image.
```

**Review by chaudum — APPROVED** [review]
The changelog entry could clarify the reason for making the parameter optional. Otherwise :rocket:

**Pull request merged**